osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,137 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+QueryData genUserAssist(QueryContext& context) {
+  QueryData results;
+  QueryData users;
+
+  queryKey("HKEY_USERS", users);
+
+  for (const auto& uKey : users) {
+    if (uKey.count("type") == 0 || uKey.count("path") == 0) {
+      continue;
+    }
+
+    std::string path = uKey.at("path");
+    std::string fullPath = path.append(
+        "\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\UserAssist");
+
+    QueryData user_assist_results;
+    queryKey(fullPath, user_assist_results);
+
+    for (const auto& rKey : user_assist_results) {
+      if (rKey.count("type") == 0 || rKey.count("path") == 0) {
+        continue;
+      }
+      std::string path_key = rKey.at("path");
+      std::string full_path_key = path_key.append("\\Count");
+
+      QueryData assist_results;
+      queryKey(full_path_key, assist_results);
+
+      for (const auto& aKey : assist_results) {
+        std::string subkey = aKey.at("path");
+
+        // split reg path by \Count\ to get Key values
+        std::size_t count_key = subkey.find("Count\\");
+        std::string value_key = subkey.substr(count_key);
+        std::string value_key_reg = value_key.substr(6, std::string::npos);
+
+        // Decode ROT13 sub key value
```

Comment:
I agree this feels better as a helper function somewhere.

I also wonder if we should be decoding it here, or not. I don't understand why it's rot13 in the first place.

Reply:
No one knows why Microsoft chose to encode the Userassist Registry key using ROT13
The current theory is that Microsoft didn't want users making random changes to the Userassist Registry Key so they encoded the values. But again no one knows for sure.